File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights):
        left = 0
        right = len(heights) - 1
        maximum = 0

        while left < right:
            width = right - left
            height = min(heights[left], heights[right])

            maximum = max(maximum, width * height)

            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1

        return maximum

File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.java
class Solution {
    public boolean isValidBST(TreeNode root) {
        Long min = Long.MIN_VALUE;
        Long max = Long.MAX_VALUE;
        
        return helper(root, min, max);
        
    }
    
    private boolean helper(TreeNode root, Long min, Long max) {
        if(root == null)
            return true;
        Long val = (long) root.val;
        if(val <= min || val >= max)
            return false;
        return helper(root.left, min, val) && helper(root.right, val, max);
    }
}